gqcp/include/Basis/Transformations/SimpleTransformation.hpp

```diff
@@ -0,0 +1,262 @@
+// This file is part of GQCG-GQCP.
+//
+// Copyright (C) 2017-2020  the GQCG developers
+//
+// GQCG-GQCP is free software: you can redistribute it and/or modify
+// it under the terms of the GNU Lesser General Public License as published by
+// the Free Software Foundation, either version 3 of the License, or
+// (at your option) any later version.
+//
+// GQCG-GQCP is distributed in the hope that it will be useful,
+// but WITHOUT ANY WARRANTY; without even the implied warranty of
+// MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+// GNU Lesser General Public License for more details.
+//
+// You should have received a copy of the GNU Lesser General Public License
+// along with GQCG-GQCP.  If not, see <http://www.gnu.org/licenses/>.
+
+#pragma once
+
+
+#include "Basis/Transformations/BasisTransformable.hpp"
+#include "Basis/Transformations/JacobiRotatable.hpp"
+#include "Mathematical/Representation/SquareMatrix.hpp"
+
+
+namespace GQCP {
+
+
+/*
+ *  MARK: SimpleTransformation implementation
+ */
+
+/**
+ *  A basis transformation that can be represented by a single transformation matrix.
+ * 
+ *  In general, we adopt the convention outlined in (https://gqcg-res.github.io/knowdes/spinor-transformations.html), where the new orbitals' coefficients can be found in the respective **column** of the related transformation matrix.
+ *  
+ *  This class is used as a base class for `RTransformation` and `GTransformation`, since they are both expressed using a single matrix, as opposed to `UTransformation`, which uses separate transformation coefficients for alpha- and beta- matrices. The word 'simple' is used here as an antonym for 'compound'.
+ * 
+ *  @tparam _Scalar                                 The scalar type used for a transformation coefficient: real or complex.
+ *  @tparam _DerivedTransformation                  The type of the transformation matrix that derives from this class, enabling CRTP and compile-time polymorphism.
+ */
+template <typename _Scalar, typename _DerivedTransformation>
+class SimpleTransformation:
```

Comment:
I like this API since it resembles the useful functionalities of `Simple1DM` and reduces redundant code!